lib/semipperm.gi

```diff
@@ -222,172 +222,172 @@ function(s)
   return gens;
 end);
 
-# isomorphisms
+# Isomorphism from an arbitrary inverse semigroup/monoid to a partial perm
+# semigroup/monoid, this is the fall back method
 
-#
-
-InstallMethod(IsomorphismPartialPermSemigroup, 
-"for a semigroup with generators",
-[IsSemigroup and HasGeneratorsOfSemigroup],
+InstallMethod(IsomorphismPartialPermSemigroup, "for a semigroup",
+[IsSemigroup],
 function(S)
-  local elts, iso, gens;
+  local set, iso, gens, T;
 
   if not IsInverseSemigroup(S) then 
-    return fail;
+    ErrorNoReturn("IsomorphismPartialPermSemigroup: usage,\n",
+                  "the argument must be an inverse semigroup,");
   fi;
   
-  elts:=AsSSortedList(S);
-
-  iso:=function(x)
-    local dom, y;
-    y:=InversesOfSemigroupElement(S, x)[1];
-    dom:=Set(elts*y);
-    return PartialPermNC(List(dom, y-> Position(elts, y)),
-     List(List(dom, y-> y*x), y-> Position(elts, y)));
+  set := AsSet(S);
+
+  iso := function(x)
+    local dom;
+    dom := Set(set * InversesOfSemigroupElement(S, x)[1]);
+    return PartialPermNC(List(dom, y -> Position(set, y)),
+                         List(List(dom, y -> y * x), 
+                              y -> Position(set, y)));
   end;
```

Comment:
This is annoying to read. As far as I can tell, you haven't actually changed anything in this part of the patch, just renamed variables and moved code around?